File: tournaments/differentSquares/differentSquares.cs
int differentSquares(int[][] matrix) {
    var store = new HashSet<int>();
    var rows = matrix.Length;
    var cols = matrix[0].Length;
    var result = 0;
    
    if (1 == rows || 1 == cols) {
        return result;
    }
    
    for (int i = 1; rows > i; ++i) {
        for (int j = 1; cols > j; ++j) {
            var temp = matrix[i - 1][j - 1] * 1000;
            temp += matrix[i - 1][j] * 100;
            temp += matrix[i][j - 1] * 10;
            temp += matrix[i][j];
            
            if (!store.Contains(temp)) {
                store.Add(temp);
                ++result;
            }
        }
    }
    
    return result;
}

File: tournaments/differentSquares/differentSquares.js
function differentSquares(matrix) {
    const store = new Set();

    if (1 === matrix.length && 1 === matrix[0].length)
        return 0;

    for (let i = 1; i in matrix; ++i) {
        for (let j = 1; j in matrix[0]; ++j) {
            let hash = matrix[i - 1][j - 1] * 1000;
            hash += matrix[i - 1][j] * 100;
            hash += matrix[i][j - 1] * 10;
            hash += matrix[i][j];

            store.add(hash);
        }
    }

    return store.size;
}
